### PR TITLE
ProducerSideService and CaptureEventProducer now support CaptureOptions

### PR DESCRIPTION
**Pull request by dpallotti**
Bug: http://b/179461890

Also performed a simple test using the Vulkan layer.

### CODE REVIEW

File: src/Service/ProducerSideServiceImpl.cpp [hunk 1]
```diff
@@ -53,6 +54,7 @@ void ProducerSideServiceImpl::OnCaptureStopRequested() {
     }
     LOG("About to send CaptureFinishedCommand to CaptureEventProducers (if any)");
     service_state_.capture_status = CaptureStatus::kCaptureFinished;
+    service_state_.capture_options = std::nullopt;
```
Comment (florian-kuebler) [inline src/Service/ProducerSideServiceImpl.cpp:57]: Is there a concrete reason why you clear the options here?

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks!

File: src/Service/ProducerSideServiceImpl.cpp (hunk 1 above)
Comment (dpallotti) [reply src/Service/ProducerSideServiceImpl.cpp:57]: Hmm, not really, it just felt like the right thing to do, as they are only kept while they could actually be used (i.e., while `service_state_.capture_status != CaptureStatus::kCaptureFinished`).

**Review by dimitry- — APPROVED** [review]

File: src/Service/ProducerSideServiceImpl.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/Service/ProducerSideServiceImpl.cpp:57]: Okay, fine for me. Thanks!

**Pull request merged**